### DISCUSSION TITLE
Always null value in persisted data

### DISCUSSION THREAD

**Discussion by AndresdoSantos** [pmndrs/zustand#1513] [Q&A]
Hello guys

I wanted to ask a question, I created this hook to get the token.

I'm using React Native

```
import create from 'zustand'
import { persist } from 'zustand/middleware'

interface Props {
  token: string | null | undefined
  getToken(item: string | null | undefined): void
}

export const useToken = create<Props, [['zustand/persist', Props]]>(
  persist(
    (set) => ({
      token: null,
      getToken: (token: string) =>
        set({
          token,
        }),
    }),
    {
      name: 'token-storage',
    },
  ),
)
```

I put the types as: `string` | `null` | `undefined`

`string` - when I have the token

`null` - when zustand is loading the token

`undefined` - when token is missing

The problem is that when I go to use the hook it just returns the null value

I'm using it like this

```
export default function App() {
  const { token } = useToken()
  console.log(token)

  if (token === null) {
    return <Loading />
  }

  if (typeof token === 'undefined') {
    return <OnBoarding />
  }

  return (
    <>
      {token && <Routes />}
    </>
  )
}
```

I expected that useToken would first return a null value but then (after the token is loaded) a string value

But what happens is that token value is always null

**Does anyone know what I'm doing wrong?**

**Comment by dai-shi** [comment]
By default, `persist` uses `localStorage`. For RN apps, you want to use `AsyncStorage`. Not sure if that's the root cause.

BTW, you shouldn't manually type store/hook with middleware, instead follow this style:
```diff
- export const useToken = create<Props, [['zustand/persist', Props]]>(
+ export const useToken = create<Props>()(
```
